packages/viewer/src/components/viewer-markup/viewer-markup.tsx

```diff
@@ -351,16 +362,36 @@ export class ViewerMarkup {
    * @ignore
    */
   @Watch('startLineAnchorStyle')
-  protected handleStartLineAnchorStyleChanged(): void {
+  protected async handleStartLineAnchorStyleChanged(): Promise<void> {
     this.updatePropsOnMarkupTool();
+
+    // If a piece of arrow markup is currently selected, then update its anchor style
+    const markup = await this.getMarkupElements();
+    markup.forEach((m) => {
+      if (m.id === this.selectedMarkupId) {
+        (m as HTMLVertexViewerMarkupArrowElement).startLineAnchorStyle =
+          this.startLineAnchorStyle;
+        this.markupChanged.emit(m);
+      }
+    });
   }
 
   /**
    * @ignore
    */
   @Watch('endLineAnchorStyle')
-  protected handleEndLineAnchorStyleChanged(): void {
+  protected async handleEndLineAnchorStyleChanged(): Promise<void> {
     this.updatePropsOnMarkupTool();
+
+    // If a piece of arrow markup is currently selected, then update its anchor style
+    const markup = await this.getMarkupElements();
+    markup.forEach((m) => {
+      if (m.id === this.selectedMarkupId) {
+        (m as HTMLVertexViewerMarkupArrowElement).endLineAnchorStyle =
+          this.endLineAnchorStyle;
+        this.markupChanged.emit(m);
+      }
+    });
```

Comment:
nit: We may be able to simplify the updates here, and avoid casting the element to an arrow element so we don't emit events/set properties on other types of markup. Thoughts on adding something like this?
```
private async getSelectedArrowMarkup(): Promise<HTMLVertexViewerMarkupArrowElement | undefined> {
  if (this.selectedMarkupId != null) {
    const selected = await this.getMarkupElement(this.selectedMarkupId);
    
    return isVertexViewerArrowMarkup(selected) ? selected : undefined;
  }
}

...

protected async handleEndLineAnchorStyleChanged(): Promise<void> {
  this.updatePropsOnMarkupTool();

  const selectedArrow = await this.getSelectedArrowMarkup();

  if (selectedArrow != null) {
    selectedArrow.endLineAnchorStyle = this.endLineAnchorStyle;
    this.markupChanged.emit(selectedArrow);
  }
}
```
with `handleStartLineAnchorStyleChanged` doing the same but setting the `startLineAnchorStyle`. Might be some further simplification, but mainly thinking to avoid re-writing the `forEach` logic and emitting events for other types of markup

Reply:
Good idea, updated!